pkg/commands/artifact/run.go

```diff
@@ -135,6 +137,13 @@ func NewRunner(ctx context.Context, cliOptions flag.Options, opts ...RunnerOptio
 	m.Register()
 	r.module = m
 
+	// Make a silent attempt to check for updates in the background
```

Comment:
@DmitriyLewen  - I'm not sure if this is the optimal place to hook into Scan runs to trigger the best efforts check for notices. Happy to move if somewhere more appropriate can be suggested

Reply:
I think your previous place was better.
You run the check before rootCommand (`PersistentPreRunE`) and show notifications after rootCommand (`PersistentPostRun`).

Both commands are on the same level

Reply:
I'm trying to sasify you suggestion that it only run on scan commands. If its in the root command there will need to be some inspection of the called sub command to ensure its a scan.

As it's a best efforts request, I think I should go back to it being run as part of the preRun and all commands get it so `--no-notices` goes back to being a global flag. wdyt?

Reply:
I looked at the current implementation again. Let's leave it like this. Given that we only need notifications when scanning, this would be the optimal place.
Thanks for clarifying this nuance.